### PR TITLE
Add jupyter notebook svc account and roles

### PR DESCRIPTION
**Pull request by r2d4**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2919)

### CODE REVIEW

**Comment by r2d4** [comment]
cc @lluunn

File: kubeflow/jupyter/jupyter-web-app.libsonnet [hunk 1]
```diff
@@ -87,6 +87,97 @@
...
+      },
+      roleRef: {
+        apiGroup: "rbac.authorization.k8s.io",
+        kind: "Role",
+        name: "jupyter-role",
+      },
+      subjects: [
+        {
+          kind: "ServiceAccount",
+          name: "jupyter",
```
Comment (lluunn) [inline kubeflow/jupyter/jupyter-web-app.libsonnet]: jupyter-notebook?

**Comment by lluunn** [comment]
Only the notebook in this namespace will work. 
Maybe the correct approach is let notebook controller to create the service account and attach to the notebook pod in the same namespace.

**Comment by r2d4** [comment]
Can we currently spawn notebooks outside kubeflow namespace?

File: kubeflow/jupyter/jupyter-web-app.libsonnet (hunk 1 above)
Comment (r2d4) [reply kubeflow/jupyter/jupyter-web-app.libsonnet]: fixed

**Comment by lluunn** [comment]
cc @jlewi

**Comment by lluunn** [comment]
I think for 0.5 we should restrict to kubeflow namespace.
The admission webhook will also mount the secret which only exists in kubeflow namespace.

Later in 0.6, we can provision these things (secret for gcp, service account) using profile.

So
/lgtm
/approve
/hold
In case someone disagrees

**Pull request merged**